File: Programmers/Lv.1/약수의합.py
def solution(n):
    answer = 0
    
    for i in range(1, n + 1):
        if n % i == 0:
            answer += i
    
    return answer

# def solution(num):
#     return sum([i for i in range(1, num + 1) if num % i == 0])
#     return num + sum([i for i in range(1, (num // 2) + 1) if num % i == 0])

File: Programmers/Lv.1.Kotlin/약수의합.kt
class Solution {
    fun solution(n: Int): Int {
        var answer = 0
        
        for (i in 1..n) {
            if (n % i == 0) {
                answer += i
            }
        }

        // answer = (1..n).filter {
        //     n % it == 0
        // }.sum()
        
        return answer
    }
}